### PR TITLE
DOC: sparse: update spdiags handling in `doc_string` and migration guide for sparray

### PR DESCRIPTION
**Pull request by dschult**
Update migration guide for sparray to direct folks to replace `spdiags` with `dia_array` instead of `diags_array`.

This is an easier migration path  because it is a direct replacement with no change to input data, only to function signature. Replacing with diags_array requires removing padding of diagonal data.

### CODE REVIEW

**Review by stefanv — APPROVED** [review]

**Review by tylerjereddy — APPROVED** [review]
Alright, it has a core developer approval and it is a small docs-only change with CI passing so I don't see a reason not to merge.

**Comment by tylerjereddy** [comment]
thanks

**Pull request merged**
